clippy_dev/src/lib.rs

```diff
@@ -136,6 +136,19 @@ pub fn gen_deprecated(lints: &[Lint]) -> Vec<String> {
         .collect::<Vec<String>>()
 }
 
+pub fn gen_register_lint_list(lints: &[Lint]) -> Vec<String> {
+    let pre = "    store.register_lints(&[".to_string();
+    let post = "    ]);".to_string();
+    let mut inner = lints
+        .iter()
+        .filter(|l| !(l.is_internal() || l.deprecation.is_some()))
+        .map(|l| format!("        &{}::{},", l.module, l.name.to_uppercase()))
```

Comment:
I'd also sort this list.
```suggestion
        .map(|l| format!("        &{}::{},", l.module, l.name.to_uppercase()))
        .sorted()
```